autoscaler/controllers/nodecollector/configmap.go

```diff
@@ -32,24 +32,34 @@ const (
 	logsServiceNameProcessorName = "resource/service-name"
 )
 
-func SyncConfigMap(sources *odigosv1.InstrumentationConfigList, signals []odigoscommon.ObservabilitySignal, allProcessors *odigosv1.ProcessorList,
-	datacollection *odigosv1.CollectorsGroup, ctx context.Context,
-	c client.Client, scheme *runtime.Scheme) error {
+func (b *nodeCollectorBaseReconciler) SyncConfigMap(ctx context.Context, sources *odigosv1.InstrumentationConfigList, signals []odigoscommon.ObservabilitySignal, allProcessors *odigosv1.ProcessorList,
+	datacollection *odigosv1.CollectorsGroup) error {
 	logger := log.FromContext(ctx)
 
 	processors := commonconf.FilterAndSortProcessorsByOrderHint(allProcessors, odigosv1.CollectorsGroupRoleNodeCollector)
 
-	desired, err := getDesiredConfigMap(ctx, sources, signals, processors, datacollection, scheme, c)
+	if b.autoscalerDeployment == nil {
+		// we only need to get the autoscaler deployment once since it can't change while this code is running
+		// (since we are running in the autoscaler pod)
+		autoscalerDeployment := &appsv1.Deployment{}
+		err := b.Client.Get(ctx, client.ObjectKey{Namespace: env.GetCurrentNamespace(), Name: k8sconsts.AutoScalerDeploymentName}, autoscalerDeployment)
+		if err != nil {
+			return err
+		}
+		b.autoscalerDeployment = autoscalerDeployment
+	}
```

Comment:
nit: can we do it once in `main` maybe and the use the deployment as a parameter while setting up manager?